test.sh

```diff
@@ -7,52 +7,131 @@ GREEN="\e[32m"
+    dcompose build &> test.log;
+}
 
+function reset_container() {
+    echo -e "resetting container ${CONTAINER_TO_BE_RESETTED};"
+    dcompose stop -t 0 ${CONTAINER_TO_BE_RESETTED} &> test.log;
+    dcompose rm -f ${CONTAINER_TO_BE_RESETTED} &> test.log;
+    dcompose build ${CONTAINER_TO_BE_RESETTED} &> test.log;
+}
 
-# application tests
-echo -e "${MAGENTA}-------------------"
-echo -e "Application checks:"
-echo -e "-------------------${RESET}"
-docker-compose run $WEBAPP_CONTAINER /bin/bash -c "python3 manage.py check"
+function dcompose() {
+    docker-compose -f ${COMPOSE_FILE} "${@}";
+}
+
+
+function run_tests() {
+    # start clean
+    reset_containers;
+    application_tests;
+    docker_volume_configuration_tests;
+    persisting_database_data_tests;
+    echo -e "cleaning up all containers"
+    reset_containers;
+}
+
+function application_tests() {
+    # application tests
+    echo -e "${MAGENTA}-------------------"
+    echo -e "Application checks:"
+    echo -e "-------------------${RESET}"
+    dcompose run $WEBAPP_CONTAINER /bin/bash -c 'python3 manage.py check' &> test.log
+    if [ $? -eq 0 ]; then
+        echo -e "${GREEN}Checks passed successfully.${RESET}"
+    else
+        echo -e "${RED}Checks failed. Please have a look at the test.log!${RESET}"
+    fi
+
+    echo -e "${MAGENTA}"
+    echo -e "------------------"
+    echo -e "Application tests:"
+    echo -e "------------------${RESET}"
+    dcompose run $WEBAPP_CONTAINER /bin/bash -c "python3 manage.py test" &> test.log
+    if [ $? -eq 0 ]; then
+        echo -e "${GREEN}Tests passed successfully.${RESET}"
+    else
+        echo -e "${RED}Tests failed. Please have a look at the test.log!${RESET}"
+    fi
+
+}
+
+function docker_volume_configuration_tests() {
+    # docker volume configuration tests
+
+    echo -e "${MAGENTA}"
+    echo -e "-------------------------"
+    echo -e "Volume integration tests:"
+    echo -e "-------------------------${RESET}"
 
+    dcompose run $CELERY_CONTAINER /bin/bash -c "touch $TEST_FILE" &> test.log
+    if [ $? -ne 0 ]; then
+        echo -e "${RED}Test file creation failed ${RESET}"
+    fi
 
-echo -e "${MAGENTA}"
-echo -e "------------------"
-echo -e "Application tests:"
-echo -e "------------------${RESET}"
-docker-compose run $WEBAPP_CONTAINER /bin/bash -c "python3 manage.py test"
+    dcompose run $WEBAPP_CONTAINER /bin/bash -c "if [ ! -f $TEST_FILE ]; then exit 1; else exit 0; fi;" &> test.log
+    if [ $? -eq 0 ]; then
+        echo -e "${GREEN}Shared test file found: volume mount correct ${RESET}"
+    else
+        echo -e "${RED}Test file does not exist: volume mount incorrect ${RESET}"
+    fi
 
+    dcompose run $CELERY_CONTAINER /bin/bash -c "rm $TEST_FILE" &> test.log
+    if [ $? -ne 0 ]; then
+        echo -e "${RED}Test file clean up failed ${RESET}"
+    fi
+}
 
-# docker volume configuration tests
+function persisting_database_data_tests() {
+    # clean containers needed for testing this...
```

Comment:
Please implement a set_up and tear_down instead. Test should not need to know if the container was used before...